### DISCUSSION TITLE
Flowrate calibration / extruder rotational distance

### DISCUSSION THREAD

**Discussion by xblax** [xblax/flashforge_ad5m_klipper_mod#70] [General]
Did calibration the extrusion on my ADM5. It seems to be the case that the default configured rotational_distance for the extruder is quite inaccurate:

```
[extruder]
rotation_distance: 4.7
```

I tested with both PLA and PETG at different volumetric flow rates. 
For 100mm requested extrusion, only about 92-93mm of filament is actually extruded. That difference from the supposed value seems to be too much to be considered as variance of a single printer.

Would be interesting if some can do measurements on their printer so that we configure a better default. Based on my measurements, a rotational distance of 4.35 would be correct.

I do also get slightly nicer first layers with the calibrated flow-rate.

**Comment by KaruroChori** [comment]
Are you confident about the diameter of the tested filament?
What was the target flow for your tests?

I printed quite a lot of parts on the very first version of this mod, and I never encountered any visible sign of under-extrusion.
I will surely test it once I am back.

**Comment by xblax** [comment]
My filament is around 1.74mm according to my caliper. But a variance in Filament would get adjusted in a filament profile in the slicer. Extrusion length however, should be more or less independed from the filament. 

I tested different flow rates between 2 and 25mm³/s with the 0.4mm nozzle. Extrusion does fall off a bit at 25, but not much. Still good enough for infill. Up to 20 it's pretty linear and I don't see much difference between PLA and PETG.

Under-extrusion isn't always easy to see. But the first layer does not get squished as nicely and the layer binding might be wore. It's also bad for water-tight prints, obviously.

I have the 0.25mm nozzle, maybe I should test with that as well.

**Reply by KaruroChori** [reply]
Ok, so you directly measured the length extruded, not just the weight of the extruded material.
I misunderstood your methodology.

Have you used some specific gcode to perform these tests?
If so, could you share it? This way we can be sure our results are aligned in methodology.

**Reply by xblax** [reply]
I do just use the extrusion tool in Mainsail. 

I measure 11cm of filament, reference point is the top of the extruder unit and mark that 11cm point with a piece of tape. Then I request extrusion of 10cm. After extrusion the tape should be 1cm away from the reference point, but it actually is ~18mm. I.e. only 92mm extruded for 100mm requested.

**Reply by KaruroChori** [reply]
Ok, great thanks

**Comment by Tiwatz** [comment]
I got a rotation distance of 4.4 so the default seems a bit off.

**Comment by mBlinkii** [comment]
@xblax im also getting 4.35, checkt this 3x.

**Comment by johncotten** [comment]
4.35 for me a well. Pro model if that makes a difference.

**Comment by xblax** [comment]
I did now set the default to 4.4: https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/printer_configs/printer.cfg
If I measure really exact I get 4.37. Also verified with my 0.25mm nozzle that the nozzle size does not affect the extruded distance.

I think 4.4 is good as safe value that should work for most. Probably better to have 1% under extrusion than over extrusion.

Thank you everyone for measuring this on your printers.

**Comment by ugryy** [comment]
Hi there. I saw this topic and extruded 100m filament extruded at 12mm^3 flow, 97mm was extruded. Then I wanted to make sure I am doing things right and found a topic on klipper3d.org. They say "It is important to use the slow extrusion rate for this test as a faster rate can cause high pressure in the extruder which will skew the results. ". I tried lower and lower values, eventually tried their recommended values. Result was getting closer to the 100mm. At 2.4mm^3 flow (G1 E100 F60), result was very very close to the 100mm. 

https://www.klipper3d.org/Rotation_Distance.html#calibrating-rotation_distance-on-extruders
